src/gatherers/service-worker.js

```diff
@@ -20,34 +20,46 @@ const Gather = require('./gather');
 
 class ServiceWorker extends Gather {
   get name() {
-    return 'serviceWorkers';
+    return 'serviceWorker';
   }
 
   reloadSetup(options) {
     const driver = options.driver;
-    this.resolved = false;
 
-    this.artifactsResolved = new Promise((res, _) => {
-      driver.on('ServiceWorker.workerVersionUpdated', data => {
-        if (!this.resolved) {
-          const controlledClients =
-              ServiceWorker.getActivatedServiceWorker(data.versions, options.url);
+    this.artifactsResolved = new Promise((resolve, reject) => {
+      function onWorkerVersionUpdated(data) {
+        driver.off('ServiceWorker.workerVersionUpdated', onWorkerVersionUpdated);
 
-          this.artifact = {
-            versions: controlledClients ? [controlledClients] : []
-          };
-          this.resolved = (typeof this.artifact.versions !== 'undefined');
-          res();
+        const version = ServiceWorker.getActivatedServiceWorker(data.versions, options.url);
+        let debugString;
+        if (!version) {
+          debugString = 'No active service worker found for this origin.';
         }
-      });
+
+        resolve({
+          version,
+          debugString
+        });
+      }
+
+      driver.on('ServiceWorker.workerVersionUpdated', onWorkerVersionUpdated);
     });
   }
 
+  /**
+   * @param {string} url
+   * @return {string}
+   */
   static getOrigin(url) {
     const parsedURL = require('url').parse(url);
     return `${parsedURL.protocol}//${parsedURL.hostname}`;
   }
 
+  /**
+   * @param {!Array<!ServiceWorkerVersion>} versions
+   * @param {string} url
+   * @return {(!ServiceWorkerVersion|undefined)}
+   */
   static getActivatedServiceWorker(versions, url) {
     const origin = this.getOrigin(url);
     return versions.find(v => v.status === 'activated' && v.scriptURL.startsWith(origin));
```

Comment:
Let's match the serviceworkermanager.js from sdk

it iterates over the results from workerVersionUpdated and extracts the `origin` for each version.
https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/sdk/ServiceWorkerManager.js?q=serviceworkermanager&sq=package:chromium&dr=C&l=681


Then it grabs the `origin` of the page.

and finds the one that matches (exactly).
https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/resources/ServiceWorkersView.js?type=cs&q=serviceworker+file:%5Esrc/third_party/WebKit/Source/devtools/front_end/&sq=package:chromium&l=72